ci/fetch-assets.ps1

```diff
@@ -0,0 +1,52 @@
+param (
+    [Parameter(Mandatory=$true)]
+    [string]$RepoName,
+    [Parameter(Mandatory=$true)]
+    [string]$DeviceDetection,
+    [string]$DeviceDetectionUrl
+)
+
+if ($env:GITHUB_JOB -eq "PreBuild") {
```

Comment:
Could you add a note here referencing where the job name is?
I don't mind too much having a check like this instead of passing a parameter, but it may add a bit of fragility which should be noted.

Reply:
Actually, having thought a little more, it's probably better not to use the environment variables at all, as this makes it specific to github and cannot be run locally.
Could we change this to look at the calling script? i.e. if it is `nightly-publish-main_prebuild.ps1`. I think this is possible in PowerShell.

Reply:
Yeah, it's kinda ugly, but it's just a GitHub-specific optimization. Not sure why you would run `nightly-publish-main_prebuild.ps1` locally if it isn't needed for Python, but this check won't prevent you from doing it. Just this specific optimization won't work, and assets will get downloaded. But locally it shouldn't really matter, because you probably wouldn't run each job in a clean environment, like GitHub does, and assets downloaded in PreBuild job can be used later.

In any case, using the calling script's name is a way better idea, so I'm gonna go do that. Thanks :)

Reply:
Hmm, for it to work we'd need to replace sourcing with calling in places like this: https://github.com/51Degrees/common-ci/blob/main/.github/workflows/nightly-publish-main.yml#L87
Otherwise, the name of the calling script would be something like `/home/runner/work/_temp/63dc3e89-09a7-44c4-8a55-29ee1fc3742d.ps1`, not `nightly-publish-main_prebuild.ps1`.

What's the reason for the script to be sourced there instead of being called normally?

Reply:
Got it. That's fine if this just means it will run fine locally, but will be less optimized.
I agree it likely won't be run locally, but we still need to allow for it (and the possibility of running on another CI platform)